msal-web-sample/src/main/java/com/microsoft/azure/msalwebsample/AuthPageController.java

```diff
@@ -43,55 +39,47 @@ public ModelAndView securePage(HttpServletRequest httpRequest) throws ParseExcep
         ModelAndView mav = new ModelAndView("auth_page");
 
         setAccountInfo(mav, httpRequest);
-
         return mav;
     }
 
     @RequestMapping("/msal4jsample/sign_out")
-    public void signOut(HttpServletRequest httpRequest, HttpServletResponse response) throws ParseException, IOException {
-
+    public void signOut(HttpServletRequest httpRequest, HttpServletResponse response) throws IOException {
         httpRequest.getSession().invalidate();
 
         response.sendRedirect(AuthHelper.END_SESSION_ENDPOINT +
-                "?post_logout_redirect_uri=" + URLEncoder.encode(appHomePage, "UTF-8"));
-    }
-
-    private void setAccountInfo(ModelAndView model, HttpServletRequest httpRequest) throws ParseException {
-        IAuthenticationResult auth = getAuthSessionObject(httpRequest);
-
-        String tenantId = JWTParser.parse(auth.idToken()).getJWTClaimsSet().getStringClaim("tid");
-
-        model.addObject("tenantId", tenantId);
-        model.addObject("account", getAuthSessionObject(httpRequest).account());
+                "?post_logout_redirect_uri=" + URLEncoder.encode(authHelper.getLogoutRedirectUrl(), "UTF-8"));
     }
 
     @RequestMapping("/obo_api")
-    public ModelAndView callOboApi(HttpServletRequest httpRequest) throws Throwable {
+    public ModelAndView callOboApi(HttpServletRequest httpRequest) throws Exception {
         ModelAndView mav = new ModelAndView("auth_page");
         setAccountInfo(mav, httpRequest);
 
-        IAuthenticationResult result =  authHelper.getAuthResultBySilentFlow(httpRequest, authHelper.configuration.oboApi);
+        IAuthenticationResult result = authHelper.getAuthResultBySilentFlow(httpRequest, authHelper.configuration.apiDefaultScope);
```

Comment:
what if silent flow fail here ?

Reply:
Silent flow shouldn't fail here, as the request made it through the AuthFilter, which guarantees that there is a valid access token, otherwise it will refresh/ prompt the user. I've wrapped this call in a try-catch in case something does fail, at which point we will clear the session, and redirect to the error page.